protected/models/Beer.php

```diff
@@ -24,6 +24,81 @@ public function relations()
 		];
 	}
 
+	public function searchBeerIds($name, $style, $hop, $yeast, $malt, $lastBeerId = null)
```

Comment:
Searching for Beer was split into two database queries for multiple reason:
Unable to construct a query which would apply filters to multiple many-to-many relationships
Searching on a single hop (or yeast or malt) returned the relavent beers but didnt display the other hops in that beer
By combining beers, hops, yeast and malt into one request it made keyoffset pagination perform bizarrely

This query returns only the beerId's that we wish to display and collecting related data is done in another query in the method below.